pkg/didcomm/protocol/exchange/state/state.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	//NullState is the did-exchange protocol initial state
+	NullState = "null"
+	//InvitedState is the did-exchange protocol invited state
+	InvitedState = "invited"
+	//RequestedState is the did-exchange protocol requested state
+	RequestedState = "requested"
+	//RespondedState is the did-exchange protocol responded state
+	RespondedState = "responded"
+	//CompleteState is the did-exchange protocol completed state
+	CompleteState = "completed"
+
+	//Reqquest Types supported in Did-exchange
+	invitation = "invitation"
+	response   = "response"
+	request    = "request"
+	ack        = "ack"
+)
+
+//Context defines the DIDComm state context
+type Context struct {
+	Store storage.Store
+	State *State
+}
+
+// New creates the new context
+func New(store storage.Store, state *State) *Context {
+	return &Context{
+		Store: store,
+		State: state,
+	}
+}
+
+// DidCommState describes the internal State machine for Inviter and Invitee. Reference: https://github.com/hyperledger/aries-rfcs/blob/master/features/0023-did-exchange/chrome_2019-01-29_07-59-38.png
+type DidCommState interface {
+	//Handle accepts the didComm protocol message which contains string representation of invitation, exchange_request, exchange_response etc.
+	Handle(msg []byte)
+	//CheckState allows the user to check the state of the state machine by passing key which is concatenation of protocol, version and connection ID (for example : didexchange1.012345678900987654321)
+	CheckState(key string) string
+}
+
+//State defines a didcomm state
+type State struct {
+	MsgType      string            `json:"@type"`
+	ConnectionID string            `json:"@id"`
+	ThreadID     map[string]string `json:"~thread"`
+	requestType  string
+	protocol     string
+	version      string
+	current      string
+}
+
+//Handle accepts the didComm protocol message and then set the state based on the message
```

Comment:
"accepts the did-exchange protocol message"

Reply:
How exchange protocol will integrate with state machine how can we create the msg